dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/mapping/MapViewDeletionHandler.java

```diff
@@ -83,6 +85,24 @@ private void deleteLegendSet(LegendSet legendSet) {
     }
   }
 
+  private void deleteProgram(Program program) {
+    List<MapView> mapViews = service.findByProgram(program);
+    for (MapView mapView : mapViews) {
+      mapView.setProgram(null);
+      if (mapView.getProgramStage() != null) {
+        program
+            .getProgramStages()
+            .forEach(
+                programStage -> {
+                  if (mapView.getProgramStage().getUid().equals(programStage.getUid())) {
+                    mapView.setProgramStage(null);
+                  }
```

Comment:
This looks a bit like `forEach` abuse. If you want to set the program stage of the map-view `null` in case the program has a stage with the same ID as the map-view is using I think something like 
```java
if (mapView.getProgramStage() != null && program         .getProgramStages().contains(mapView.getProgramStage()) {
   mapView.setProgramStage(null);
}            
```
is more clear. Maybe you want to compare IDs to be more safe that contains can be written similarly using `program.getProgramStages().stream().map(IdentifiableObject::getUid).toList().containts...`